### PR TITLE
Fix HF endpoint returns blank for text-generation

### LINKED ISSUE #7353: HuggingFaceEndpoint Class Bug
### System Info

the provided class
[langchain/langchain/llms/huggingface_endpoint.py](https://github.com/hwchase17/langchain/blob/370becdfc2dea35eab6b56244872001116d24f0b/langchain/llms/huggingface_endpoint.py)
 class HuggingFaceEndpoint(LLM): 

has a bug.
it should be
```python
if self.task == "text-generation":
# Text generation return includes the starter text.
text = generated_text[0]["generated_text"]
```
not
```python
text = generated_text[0]["generated_text"][len(prompt) :]
```

the current class will likely just return a 0.

### Who can help?

_No response_

### Information

- [X] The official example notebooks/scripts
- [ ] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

```python
    llm= HuggingFaceEndpoint(endpoint_url=os.getenv('ENDPOINT_URL'),task="text-generation",
                              model_kwargs={"temperature":0.7, "max_length":512})
 conversation_chain = ConversationalRetrievalChain.from_llm(
        llm=llm,
        retriever=vectorstore.as_retriever(),
        memory=memory
    )

```

### Expected behavior

the output is 0

### PR DESCRIPTION
**Pull request by HashemAlsaket**
Description: Current `_call` function in the `langchain.llms.HuggingFaceEndpoint` class truncates response when `task=text-generation`. Same error discussed a few days ago on Hugging Face: https://huggingface.co/tiiuae/falcon-40b-instruct/discussions/51
Issue: Fixes #7353 
Tag maintainer: @hwchase17 @baskaryan @hinthornw

### CODE REVIEW

**Comment by baskaryan** [comment]
hmm im guessing for some models the text-generation endpoint does include the prompt? is there any good way to identify when that is the case?

**Comment by HashemAlsaket** [comment]
Since the response based on model-source is unpredictable, I think checking if the prompt is included is a good choice.

Example:
`gpt-2` retains the prompt in the returned response. Quick string compare is used to remove it:

`
generated text ------------------------------------------------------------------------------
[{'generated_text': 'Question: When does summer begin?\n\nAnswer: \xa0April 27. Some time in March. \nIn short, you should be able to have a little fun out on your patio by the end of March. I also recommend taking a trip'}]

remove prompt from response -------------------------------------------------------------

April 27. Some time in March.
In short, you should be able to have a little fun out on your patio by the end of March. I also recommend taking a trip
`

**Comment by baskaryan** [comment]
> Since the response based on model-source is unpredictable, I think checking if the prompt is included is a good choice.
> 
> Example: `gpt-2` retains the prompt in the returned response. Quick string compare is used to remove it:
> 
> ```
> generated text ------------------------------------------------------------------------------
> [{'generated_text': 'Question: When does summer begin?\n\nAnswer: \xa0April 27. Some time in March. \nIn short, you should be able to have a little fun out on your patio by the end of March. I also recommend taking a trip'}]
> 
> remove prompt from response -------------------------------------------------------------
> 
> April 27. Some time in March.
> In short, you should be able to have a little fun out on your patio by the end of March. I also recommend taking a trip
> ```

i like that idea!

**Comment by HashemAlsaket** [comment]
@baskaryan good to push?

**Comment by baskaryan** [comment]
done, thanks @HashemAlsaket!
